### PR TITLE
kotlintree on CPG demo

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- added CPG demo using tree-sitter
- added choice of engine on FE
- updated query

demo-cpg-tree-sitter

### CODE REVIEW

File: save-demo-cpg/build.gradle.kts [hunk 1]
```diff
@@ -20,6 +20,7 @@ repositories {
         }
     }
     mavenCentral()
+    mavenLocal()
```
Comment (nulls) [inline save-demo-cpg/build.gradle.kts:23]: a temporary till `kotlin-tree-jna` is released to github packages

File: save-demo-cpg/src/main/kotlin/com/saveourtool/save/demo/cpg/controller/CpgController.kt [hunk 2]
```diff
@@ -43,31 +45,56 @@ const val FILE_NAME_SEPARATOR = "==="
...
                 .map {
-                    cpgRepository.save(it)
+                    saveFunction(it)
                 }
-                .map {
+                .map { queryId ->
                     CpgResult(
-                        cpgRepository.getGraph(it),
+                        graphFunction(queryId),
                         "match (e: Component where e.name = \"${tmpFolder.fileName.name}\") return e;",
```
Comment (nulls) [inline save-demo-cpg/src/main/kotlin/com/saveourtool/save/demo/cpg/controller/CpgController.kt:98]: better to replace by a query using `queryId`

File: save-demo-cpg/build.gradle.kts (hunk 1 above)
Comment (nulls) [reply save-demo-cpg/build.gradle.kts]: added a link to fork

**Review by 0x6675636b796f75676974687562 — APPROVED** [review]

File: save-demo-cpg/build.gradle.kts (hunk 1 above)
Comment (nulls) [reply save-demo-cpg/build.gradle.kts:23]: need to remove mavenLocal when artifacts will be ready

**Review by sanyavertolet — APPROVED** [review]

**Review by Cheshiriks — APPROVED** [review]

File: save-demo-cpg/build.gradle.kts (hunk 1 above)
Comment (0x6675636b796f75676974687562) [reply save-demo-cpg/build.gradle.kts]: Removed with 666848983427d480af4983e71a28903b648e8e3d.

**Pull request merged**